bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -2193,6 +2193,9 @@ private static boolean checkFillerValue(BUnionType type) {
         if (type.isNullable()) {
             return true;
         }
+        if (type.getMemberTypes().isEmpty()) {
```

Comment:
I'm not sure if this is the correct fix. IMO, we should only be creating a union type if there's at least two members. So ideally we wouldn't have a union with no members.

This is the approach we usually take in the compiler too. See https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L3831